### PR TITLE
feat: support container socket via env variable

### PR DESCRIPTION
**Pull request by madpah**

### CODE REVIEW

File: src/main/java/com/neuvector/ScannerEnvironmentImpl.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package com.neuvector;
+
+public class ScannerEnvironmentImpl implements ScannerEnvironment {
+
+    private String containerSocket;
+
+    @Override
+    public void create() {
+        String containerRuntimeSocket = System.getenv(ScannerEnvironment.ENV_CONTAINER_RUNTIME_SOCKET);
```
Comment (koraytugay) [inline src/main/java/com/neuvector/ScannerEnvironmentImpl.java:9]: We should not use System.getenv but rather inject this value from insight-scanner as we do for similar values. System.getenv does not work in Jenkins.

File: src/test/java/com/neuvector/ScannerTest.java [hunk 2]
```diff
@@ -1,32 +1,33 @@
...
+
 /**
  * Unit tests
  */
 public class ScannerTest 
 {
-    @Rule
-    public TemporaryFolder mountFolder= new TemporaryFolder();
+    @TempDir
+    public File mountFolder;
```
Comment (koraytugay) [inline src/test/java/com/neuvector/ScannerTest.java:30]: Why these changes?

File: src/test/java/com/neuvector/ScannerTest.java [hunk 3]
```diff
@@ -41,7 +42,7 @@ public void scanLocalImageTest() throws IOException {
         String nvRegistryPassword = null;
         String nvRegistryURL = "https://registry.hub.docker.com";
         //mountPath is an optional parameter. It will use "/var/neuvector" by default.
-        String mountPath = mountFolder.getRoot().getAbsolutePath();
+        String mountPath = mountFolder.getAbsolutePath();
```
Comment (koraytugay) [inline src/test/java/com/neuvector/ScannerTest.java:45]: Reason for the change? Looks unrelated to the task at hand?

File: src/test/java/com/neuvector/ScannerTest.java [hunk 4]
```diff
@@ -121,7 +122,7 @@ public void scanRegistryTest_BindMountShared() throws IOException {
         String nvRegistryPassword = null;
         boolean bindMountShared = true;
         //mountPath is an optional parameter. It will use "/var/neuvector" by default.
-        String mountPath = mountFolder.getRoot().getAbsolutePath();
+        String mountPath = mountFolder.getAbsolutePath();
```
Comment (koraytugay) [inline src/test/java/com/neuvector/ScannerTest.java:125]: Reason for change?

File: src/test/java/com/neuvector/ScannerTest.java (hunk 2 above)
Comment (madpah) [reply src/test/java/com/neuvector/ScannerTest.java:30]: Upgraded jUnit to v5 as part of this.

File: src/test/java/com/neuvector/ScannerTest.java (hunk 3 above)
Comment (madpah) [reply src/test/java/com/neuvector/ScannerTest.java:45]: Upgraded jUnit to v5 as part of this.

File: src/test/java/com/neuvector/ScannerTest.java (hunk 4 above)
Comment (madpah) [reply src/test/java/com/neuvector/ScannerTest.java:125]: Upgraded jUnit to v5 as part of this.

**Comment by madpah** [comment]
Closing - will re-raise with implementation that matches suggestion from @koraytugay.
